inst/odin/compare_carehomes.cpp

```diff
@@ -0,0 +1,215 @@
+  typedef typename T::real_t real_t;
+
+  // State variables; these largely correspond to the quantities in data
+  const real_t model_icu = odin(ICU_tot);
+  const real_t model_general = odin(general_tot);
+  const real_t model_hosp = model_icu + model_general;
+  const real_t model_deaths_comm = odin(D_comm_inc);
+  const real_t model_deaths_hosp = odin(D_hosp_inc);
+  const real_t model_admitted = odin(admit_conf_inc);
+  const real_t model_new = odin(new_conf_inc);
+  const real_t model_new_admitted = model_admitted + model_new;
+  const real_t model_sero_pos = odin(sero_pos);
+  const real_t model_sympt_cases = odin(sympt_cases_inc);
+  const real_t model_sympt_cases_over25 = odin(sympt_cases_over25_inc);
+  const real_t model_sympt_cases_non_variant_over25 =
+    odin(sympt_cases_non_variant_over25_inc);
+  const real_t model_react_pos = odin(react_pos);
+
+  // This is used over and over
+  const real_t exp_noise = odin(exp_noise);
+
+  // TODO: need sum(N_tot) here
+  const real_t N_tot_all = odin(N_tot_all); // sum(pars$N_tot)
+  const real_t pillar2_negs =
+    odin(p_NC) * (odin(N_tot_all) - model_sympt_cases);
+  const real_t model_pillar2_prob_pos =
+    test_prob_pos(model_sympt_cases,
+                  pillar2_negs,
+                  odin(pillar2_sensitivity),
+                  odin(pillar2_specificity),
+                  exp_noise,
+                  rng_state);
+
+  const real_t pillar2_over25_negs =
+    odin(p_NC) * (odin(N_tot_over25) - model_sympt_cases_over25);
+  const real_t model_pillar2_over25_prob_pos =
+    test_prob_pos(model_sympt_cases_over25,
+                  pillar2_over25_negs,
+                  odin(pillar2_sensitivity),
+                  odin(pillar2_specificity),
+                  exp_noise,
+                  rng_state);
+
+  const real_t N_tot_react = odin(N_tot_react); // sum(pars$N_tot[2:18])
+  const real_t model_react_prob_pos =
+    test_prob_pos(model_react_pos,
+                  N_tot_react - model_react_pos,
+                  odin(react_sensitivity),
+                  odin(react_specificity),
+                  exp_noise,
+                  rng_state);
+
+  // serology
+  const real_t model_sero_prob_pos =
+    test_prob_pos(model_sero_pos,
+                  odin(N_tot_15_64) - model_sero_pos,
+                  odin(sero_sensitivity),
+                  odin(sero_specificity),
+                  exp_noise,
+                  rng_state);
+
+  // Strain
+  const real_t model_strain_over25_prob_pos =
+    test_prob_pos<real_t>(model_sympt_cases_non_variant_over25,
+                          model_sympt_cases_over25 -
+                          model_sympt_cases_non_variant_over25,
+                          1, 1, exp_noise,
+                          rng_state);
+
+  // Note that in ll_nbinom, the purpose of exp_noise is to allow a
+  // non-zero probability when the model value is 0 and the observed
+  // value is non-zero (i.e. there is overreporting)
```

Comment:
Is that necessarily over-reporting? Couldn't the model be wrong?

Reply:
No idea! This is ported over from the R :) https://github.com/mrc-ide/sircovid/blob/i200-compiled-compare/R/carehomes.R#L569-L571